core/src/test/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptorTest.java

```diff
@@ -0,0 +1,152 @@
+package com.opensymphony.xwork2.interceptor;
+
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.XWorkTestCase;
+import com.opensymphony.xwork2.mock.MockActionInvocation;
+import org.apache.struts2.ServletActionContext;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import java.util.Arrays;
+import java.util.List;
+
+// TODO Vary headers tests
+
+public class CSRFPreventionInterceptorTest extends XWorkTestCase {
+
+    CSRFPreventionInterceptor interceptor;
+
+
+    public void testNoSite() throws Exception {
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        request.removeHeader("sec-fetch-site");
+
+        ServletActionContext.setRequest(request);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        try{
+            interceptor.intercept(mai);
+            assert(true);
+        } catch (RuntimeException e){
+            assert(false);
+        }
```

Comment:
let's adapt these try-catch blocks to our changes to the Interceptor, in particular checking that the response has the right response code on each case